### PR TITLE
[FIX] Selling decimals issues (Insufficient crops to sell)

### PR DESCRIPTION
**Pull request by mGranTY**
# Description

Fixes an error that happens when a user tries to sell a crop with all the 18 decimals places. This is common for users with the green amulet

BEFORE:

https://user-images.githubusercontent.com/94913189/196090883-24647934-030a-41a3-8267-847278847490.mp4

AFTER:

https://user-images.githubusercontent.com/94913189/196090979-64614d2f-2528-40c3-adf0-601714dabeaa.mp4

## THIS NEEDS EXTENSIVE TESTING AS IT CHANGES THE SELL FLOW!!! 

idk if it needs a backend change.

## Type of change

- [X] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# How Has This Been Tested?

yarn test
image

yarn test sell.test
image

# Checklist:

- [X] Title of the PR is relevant and is prefixed with [FEAT], [CHORE] or [FIX]
- [ ] Screenshot if it includes any UI changes
- [X] I have read the contributing guidelines and agree to the T&Cs
- [X] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/features/game/lib/constants.ts [hunk 1]
```diff
@@ -468,6 +468,7 @@ export const INITIAL_FARM: GameState = {
   inventory: {
     Sunflower: new Decimal(5),
     Potato: new Decimal(12),
+    Carrot: new Decimal("502.079999999999991"),
```
Comment (hujikol) [inline src/features/game/lib/constants.ts:471]: did you have tried out to use pure 0.9 floating points? i.e. `502.999999999999999` it's 15 - 9 right there

Comment (mGranTY) [reply src/features/game/lib/constants.ts:471]: works just fine. Only the modal shows rounded up, but the actual value is the one sold.

https://user-images.githubusercontent.com/94913189/196141315-ed9dc3a8-f93d-47f9-a4b7-63cd0fe5a769.mp4

**Comment by mGranTY** [comment]
now the sell modal is precise, maybe we can do a toFixed to only show idk 2-4 zeros?

image

**Comment by adamhannigan** [comment]
@mGranTY Yes, we typically fix to 4 places. That would be ideal in this modal flow 😄

**Review by adamhannigan — APPROVED** [review]
Great addition! 🏆 Well done

**Pull request merged**